client/gutenberg/editor/browser-url/index.js

```diff
@@ -33,6 +35,11 @@ export class BrowserURL extends Component {
 		) {
 			this.props.replaceHistory( `${ currentRoute }/${ postId }` );
 		}
+
+		if ( postStatus === 'trash' && endsWith( currentRoute, `/${ postId }` ) ) {
+			const postType = startsWith( currentRoute, '/gutenberg/post/' ) ? 'posts' : 'pages';
```

Comment:
You can get the post type from the state and pass it as a prop. The `edit-post` package is doing something similar: https://github.com/WordPress/gutenberg/blob/471e581f0467adaf823dbfabee080ba050a2118f/packages/edit-post/src/components/browser-url/index.js#L96

Reply:
Also please consider the CPT.
Their path format is something like this:
`/types/feedback` for the post list
`/edit/feedback` for the editor (or, of course, `/gutenberg/edit/feedback` for Guten)

Reply:
I kind of ignored other post types. This was more of a quick fix than a something else. I've updated the checks on dff24b9f58a2de80dfed43fbc5ee5da57d514c6d.